internal/handler/user_service.go

```diff
@@ -30,25 +39,31 @@ func (s *UserService) AddUser(ctx context.Context, addUser *pb.AddUserRequest) (
 		Address:  addUser.Address,
 		Username: addUser.Username,
 		Phone:    addUser.Phone}
+	s.mtx.Lock()
```

Comment:
Golang provides more suitable kind of mutexes for such operations. RWMutex allows to guard different kinds of the operations: read and write. It might be more effective in cases when we have a lot of read requests, but from time to time we will need to update the data. Probably this code also can be improved

Reply:
Done